src/serializers/omitPropertiesSerializer.ts

```diff
@@ -1,38 +1,19 @@
 import { omitProperties } from './omitProperties';
+import type { SerializerFn } from './types';
 
-export interface OmitPropertiesSerializerOptions {
+export const createOmitPropertiesSerializer = (
   /**
    * A list of properties that should not be logged.
    */
-  omitPropertyNames: string[];
-}
-
-type OmitHeaderNamesFn = (record: unknown) => unknown;
-type OmitHeaderNamesSerializer = Record<string, OmitHeaderNamesFn>;
+  properties: string[],
+): SerializerFn => {
+  const uniquePropertySet = new Set(properties);
 
-/** Creates a serializer that operates on the logged object's top-level property named `topLevelPropertyName`
- *  and omits the properties listed in `options.omitPropertyNames`.
- *  @param topLevelPropertyName - The name of the root property on the logged object that to omit properties from.
- *  @param options - Options for the serializer.
- */
-export const createOmitPropertiesSerializer = (
-  topLevelPropertyName: string,
-  options: OmitPropertiesSerializerOptions,
-): OmitHeaderNamesSerializer => {
-  const propertyNames = [...new Set(options.omitPropertyNames ?? [])].filter(
-    (propertyName) => typeof propertyName === 'string',
-  );
-
-  if (
-    !topLevelPropertyName ||
-    typeof topLevelPropertyName !== 'string' ||
-    topLevelPropertyName.length === 0 ||
```

Comment:
My thinking is that we don't need to be quite this defensive, because the only variable input to `createOmitPropertiesSerializer` is `omitHeaderNames`, and that should be statically defined and type checked in the consumer repo.

By contrast it makes sense to be defensive when serializing an arbitrary logging object as we don't have any type guarantees around that.

Reply:
I don't mind this change since as you say they're meant to be statically defined and type-checked.

_My thoughts were that it's done on initialisation and cleansing external inputs is just automatic (since you can cast `as unknown as ExpectedType`) so wasn't expensive anyway._